### PR TITLE
fix: nightly e2e fixes

### PR DESCRIPTION
**Pull request by effects-ai**
Fixes for the Nightly E2E tests.

### CODE REVIEW

File: .github/action_scripts/delegated_staking/token-lock-replacement-edge-cases.js [hunk 1]
```diff
@@ -565,13 +573,29 @@ const cleanupExistingStakes = async (urls, account) => {
...
       maxStalledChecks: 120,
       interval: 5000,
     },
   )
-  logWorkflow.info('Pre-existing stakes cleaned up, token locks unlocked')
+
+  logWorkflow.info('Pre-existing stakes and token locks cleaned up')
 }
 
 /**
```
Comment (ryle-ai) [inline .github/action_scripts/delegated_staking/token-lock-replacement-edge-cases.js:611]: If `initialLocks.length > 0` but `active.length === 0 && pending.length === 0` (orphaned locks from a previously-interrupted cleanup), no withdrawals are issued and the lock poll below will spin for the full `maxOrdinalMisses: 60` budget (~43 minutes) before failing. This is the expected failure mode on the first post-deploy run if old locks are still present — worth being aware of if the nightly run goes red immediately after merge.

**Review by ryle-ai — COMMENTED** [review]
The fix is correct. The docstring accurately describes the race: pending withdrawals can reach zero before the corresponding TokenUnlock events are processed out of `currentTokenLocks`, so a second independent poll for locks is the right approach.

One edge case: if a previous run was interrupted after stakes fully drained (active=0, pending=0) but before locks cleared — most likely on the very first nightly run after this fix is deployed, since the old code had no lock-drain poll — the next run will see `active=0, pending=0, initialLocks=N`. No withdrawals are issued, the stakes poll passes immediately, and the locks poll spins for up to ~43 minutes (60 ordinal progressions × ~43s) waiting for locks that have no in-flight withdrawal to release them. This results in a noisy CI failure rather than a silent one, so it's not a blocker, but the first post-deploy run may need manual intervention if the old locks haven't cleared yet.

PR description is empty.

**Review by ryle-ai — COMMENTED** [review]
Both fixes are correct and address real nightly E2E failures.

**Token lock cleanup:** The two-phase poll (stakes first, then locks) correctly handles the race where `pendingWithdrawals` reaches zero before the corresponding `TokenUnlock` events clear from `currentTokenLocks`. The `withRetryOrdinal` contract is used correctly — the new locks poll is independent of the stakes poll, which is exactly what the race requires.

One edge case to be aware of for the first post-deploy run: if `initialLocks > 0` but `active == 0` and `pending == 0` (orphaned locks from a prior interrupted run where old code left locks behind), the function skips the early-return, issues no withdrawals, immediately passes the stakes poll, then spins on the locks poll waiting for locks that no in-flight withdrawal will ever unlock. After 60 ordinal progressions (~43 minutes), `withRetryOrdinal` throws `"Failed after 60 ordinal progressions. Transaction likely dropped or invalid. Last error: Still N active token locks"` — which is a misleading message since no transaction was submitted. This results in a noisy CI failure rather than the previous silent accumulation, which is strictly better, but operators seeing this error the first time may spend time looking for a dropped transaction that doesn't exist.

**Snapshot-streaming postgres startup:** Verified against `docker/snapshot-streaming/docker-compose.yaml` in develop — `container_name: snapshot-streaming-postgres` is explicitly set, the volume is `ss-pgdata`, and the DATABASE_URL credentials match the compose environment exactly. The TCP readiness check (`-h 127.0.0.1`) correctly avoids the unix-socket race described in the comment. The sequence (postgres up → TCP ready → schema push → app up) is correct.

PR description is empty.

**Review by ryle-ai — CHANGES_REQUESTED** [review]
Two issues, one of which should be cleaned up before merge.

**PR description is empty.** Please document what broke, why it broke, and what each fix does.

**`push: nighty_e2e_fixes` trigger must be removed before merge.** The trigger was added to make it easy to test the workflow changes during PR development, but it shouldn't land in `develop`. The `nightly-rebuild` and `nightly-deploy` workflows share the `nightly-deploy` concurrency group with `cancel-in-progress: true`, so any push to `nighty_e2e_fixes` while a legitimate nightly run is in progress will silently cancel it. After the branch is deleted post-merge the trigger becomes dead config, but it's still messy and represents a real cancellation risk during development. Remove the `push:` block and keep only `schedule` + `workflow_dispatch`.

**Orphaned-locks edge case (non-blocking, first-deploy awareness).** When `active === 0 && pending === 0 && initialLocks > 0` — the state left by the old code on any prior run that exited cleanly — the new code correctly skips the early return, runs the stakes poll (passes immediately), then spins on the locks poll for up to 60 ordinal progressions (~43 minutes) waiting for locks that have no in-flight withdrawal to release them. The error thrown at timeout says `cleanupExistingStakes:tokenLocks: Failed after 60 ordinal progressions. Transaction likely dropped or invalid. Last error: Still N active token locks`, which is misleading since no transaction was submitted. This is a noisy CI failure rather than silent accumulation, so it's strictly better than before, but the first nightly run after merge may need manual intervention.

**Review by ryle-ai — CHANGES_REQUESTED** [review]
The two fixes are correct. The stakes/locks two-phase poll addresses a real race and the postgres startup sequencing is sound. One item needs to be removed before merge.

**`push: nighty_e2e_fixes` trigger must be removed.** `nightly-rebuild.yml` and `nightly-deploy.yml` share the `nightly-deploy` concurrency group with `cancel-in-progress: true`. Any push to `nighty_e2e_fixes` while a nightly deploy or rebuild is in progress silently cancels it. The branch will be deleted after merge so the trigger becomes dead config, but it can cancel legitimate nightly runs during development and during the window between merge and branch deletion.

**Orphaned-locks edge case (non-blocking).** When `active === 0 && pending === 0 && initialLocks > 0` — state left by any prior run that exited cleanly before this fix — the stakes poll passes immediately, then the locks poll spins for up to 60 ordinal progressions (~43 minutes) waiting for locks that have no in-flight withdrawal to release them. `withRetryOrdinal` then throws `cleanupExistingStakes:tokenLocks: Failed after 60 ordinal progressions. Transaction likely dropped or invalid. Last error: Still N active token locks`, which is misleading because no transaction was submitted. The first nightly run after merge may fail this way if old locks are still present.

**`proofsHash` comment missing (non-blocking).** `compose-runner.sh` (the existing local seed-creation path this mirrors) includes an explanatory comment: `"The node uses Brotli-compressed JSON for hashing, which we cannot replicate here. This simplified hash is sufficient for the E2E test since snapshot-streaming does not cryptographically verify the seed proofsHash."` The nightly-rebuild version of the same code omits this, which will puzzle a future reader wondering why the hash is computed this way.

**Empty PR description.**

**Review by scasplte2 — APPROVED** [review]
